packages/dd-trace/src/agentless/exporter.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+const log = require('../log')
+const format = require('../format')
+
+const MAX_SIZE = 255 * 1024 // 255kb
+
+class LogExporter {
+  constructor (outputStream, maxSize = MAX_SIZE) {
+    this._outputStream = outputStream
+    this._maxSize = maxSize
+  }
+
+  export (span) {
+    const spanContext = span.context()
+    const trace = spanContext._trace
+
+    log.debug(() => `Adding trace to queue: ${JSON.stringify(trace)}`)
+
+    let size = 0
+    let queue = []
+
+    for (const span of trace.finished) {
```

Comment:
This should not happen on every export.

Reply:
Isn't the point of the LogExporter to flush whenever it has a finished span?

Reply:
If you want to send individual spans as they are finished, why is the loop needed? You should just have to send the span directly. Also, I thought the idea was to batch multiple spans as a single log entry.

Reply:
So, the processor, (which calls the exporters), won't try and send a span until the trace it's on has all it's spans completed, which was originally logic that was in the Writer. We don't want to have one span per log line, because it's going to cause a lot of invokes of our log forwarder and be expensive for our customers. Maybe the exporters should take a trace as input instead?

Reply:
So, what I ended up settling on doing, is sending the exporters a list of spans they need to write out. The processor will only send a list of spans to be exported once the trace is complete.

Reply:
This should be done at the exporter level. Basically the processor sends every span to the exporter and each exporter can then have their own logic on how to buffer them. This will be useful if we want to change the strategy in a single exporter, for example to stream spans.